packages/features/ee/workflows/lib/reminders/emailReminderManager.ts

```diff
@@ -136,200 +116,30 @@ const scheduleEmailReminderForEvt = async (args: scheduleEmailReminderArgs & { e
     action,
   } = args;
 
-  const { startTime, endTime } = evt;
   const uid = evt.uid as string;
-  const timeUnit: timeUnitLowerCase | undefined = timeSpan.timeUnit?.toLocaleLowerCase() as timeUnitLowerCase;
-
-  let scheduledDate = null;
-
-  if (triggerEvent === WorkflowTriggerEvents.BEFORE_EVENT) {
-    scheduledDate = timeSpan.time && timeUnit ? dayjs(startTime).subtract(timeSpan.time, timeUnit) : null;
-  } else if (triggerEvent === WorkflowTriggerEvents.AFTER_EVENT) {
-    scheduledDate = timeSpan.time && timeUnit ? dayjs(endTime).add(timeSpan.time, timeUnit) : null;
-  }
-
-  let attendeeToBeUsedInMail: AttendeeInBookingInfo | null = null;
-  let name = "";
-  let attendeeName = "";
-  let timeZone = "";
-
-  switch (action) {
-    case WorkflowActions.EMAIL_ADDRESS:
-      name = "";
-      attendeeToBeUsedInMail = evt.attendees[0];
-      attendeeName = evt.attendees[0].name;
-      timeZone = evt.organizer.timeZone;
-      break;
-    case WorkflowActions.EMAIL_HOST:
-      attendeeToBeUsedInMail = evt.attendees[0];
-      name = evt.organizer.name;
-      attendeeName = attendeeToBeUsedInMail.name;
-      timeZone = evt.organizer.timeZone;
-      break;
-    case WorkflowActions.EMAIL_ATTENDEE: {
-      // check if first attendee of sendTo is present in the attendees list, if not take the evt attendee
-      const attendeeEmailToBeUsedInMailFromEvt = evt.attendees.find(
-        (attendee) => attendee.email === sendTo[0]
-      );
-      attendeeToBeUsedInMail = attendeeEmailToBeUsedInMailFromEvt
-        ? attendeeEmailToBeUsedInMailFromEvt
-        : evt.attendees[0];
-      name = attendeeToBeUsedInMail.name;
-      attendeeName = evt.organizer.name;
-      timeZone = attendeeToBeUsedInMail.timeZone;
-      break;
-    }
-  }
 
-  let emailContent = {
+  const scheduledDate = WorkflowService.processWorkflowScheduledDate({
+    workflowTriggerEvent: triggerEvent,
+    time: timeSpan.time,
+    timeUnit: timeSpan.timeUnit,
+    evt,
+  });
+  const workflowReminderRepository = new WorkflowReminderRepository(prisma);
+  const bookingSeatRepository = new BookingSeatRepository(prisma);
+  const emailWorkflowService = new EmailWorkflowService(workflowReminderRepository, bookingSeatRepository);
```

Comment:
DI setup. In the future we can look at creating a DI container for the `EmailWorkflowService`